doc/source/dev/gitwash/development_workflow.rst

```diff
@@ -16,93 +16,63 @@ Basic workflow
-         git fetch upstream
-	 git merge --no-ff upstream/master
-
-         # Recheck that what is there is sensible
-         git log --oneline --graph
-         git log -p upstream/master..
-
-         # Finally, push branch to upstream master
-	 git push upstream my-new-feature:master
-
-     See :ref:`below <pushing-to-main>`.
-
-.. note:: It's usually a good idea to use the ``-n`` flag to ``git push``
-          to check first that you're about to push the changes you want to
-          the place you want.
+     :ref:`create a pull request <asking-for-merging>`.
 
+   - *Core developers* If you want to push changes without
+     further review, see the notes :ref:`below <pushing-to-main>`.
+     
 This way of working helps to keep work well organized and the history
 as clear as possible.
 
-.. note::
-
-   Do not use ``git pull`` --- this avoids common mistakes if you are
-   new to Git. Instead, always do ``git fetch`` followed by ``git
-   rebase``, ``git merge --ff-only`` or ``git merge --no-ff``,
-   depending on what you intend.
-
 .. seealso::
 
-   See discussions on `linux git workflow`_,
-   and `ipython git workflow <http://mail.scipy.org/pipermail/ipython-dev/2010-October/006746.html>`__.
+   There are many online tutorials to help you `learn git`_. For discussions
+   of specific git workflows, see these discussions on `linux git workflow`_,
+   and `ipython git workflow`_.
 
 .. _making-a-new-feature-branch:
 
 Making a new feature branch
 ===========================
 
-To update your master branch, use::
+First, update your master branch with changes that have been made in the main
+Numpy repository. In this case, the ``--ff-only`` flag ensures that a new
+commit is not created when you merge the upstream and master branches. It is
+very important to avoid merging adding new commits to ``master``.
+
+::
 
+   # go to the master branch
+   git checkout master
+   # download changes from github
    git fetch upstream
+   # update the master branch
    git merge upstream/master --ff-only
+   # Push new commits to your Github repo
+   git push
 
-To create a new branch and check it out, use::
-
-   git checkout -b my-new-feature upstream/master
-
-Generally, you will want to keep this branch also on your public github_ fork
-of NumPy_.  To do this, you `git push`_ this new branch up to your github_
-repo.  Generally (if you followed the instructions in these pages, and
-by default), git will have a link to your github_ repo, called
-``origin``.  You push up to your own repo on github_ with::
+.. note::
 
-   git push origin my-new-feature
+   You could also use ``pull``, which combines ``fetch`` and ``merge``, as
+   follows::
+   
+        git pull upstream master
 
-In git >= 1.7 you can ensure that the link is correctly set by using the
-``--set-upstream`` option::
+   However, never use ``git pull`` without indicating the source and
+   destination, as the inherent ambiguity can cause problems. This avoids a
```

Comment:
The pair `upstream master` is the source, the destination is the current branch. If you want to be consistent with the fetch/merge version `git pull --ff-only upstream master`.